components/image_viewer/image_viewer.test.js

```diff
@@ -0,0 +1,183 @@
+import { assert } from 'chai';
+import { mount } from '@vue/test-utils';
+import DtImageViewer from './image_viewer.vue';
+
+// Constants
+const basePropsData = {
+  imageSrc: 'test.png',
+  imageAlt: 'imageAltText',
+  closeAriaLabel: 'closeButtonAriaLabel',
+  imageButtonClass: 'd-wmn64 d-hmn64 w-wmx332 d-hmx332',
+  ariaLabel: 'Click to open image',
+};
+
+describe('DtImageViewer Tests', function () {
+  // Wrappers
+  let wrapper;
+  let imageViewerPreview;
+  let imageViewerFull;
+  let closeButton;
+  let previewImage;
+  let fullImage;
+  let overlay;
+
+  // Environment
+  let propsData = basePropsData;
+  let attrs = {};
+  let slots = {};
+  let provide = {};
+
+  // Helpers
+  const _setChildWrappers = () => {
+    imageViewerPreview = wrapper.find('[data-qa="dt-image-viewer-preview"]');
+    previewImage = imageViewerPreview.find('img');
+  };
+
+  const _openModal = async () => {
+    await imageViewerPreview.trigger('click');
+    _setFullImageChildWrappers();
+  };
+
+  const _setFullImageChildWrappers = () => {
+    imageViewerFull = wrapper.find('[data-qa="dt-image-viewer-full"]');
+    fullImage = imageViewerFull.find('img');
+    closeButton = wrapper.find('[data-qa="dt-image-viewer-close-btn"');
+    overlay = wrapper.find('[data-qa="dt-modal"]');
+  };
+
+  const _setWrappers = () => {
+    wrapper = mount(DtImageViewer, {
+      propsData,
```

Comment:
in vue3 this has been renamed to just "props", might have been related to your problems.